patches/tModLoader/Terraria/ModLoader/BuilderToggle.cs

```diff
@@ -57,11 +58,21 @@ public abstract class BuilderToggle : ModTexturedType, ILocalizedModType
 	/// <value>Default value is 2</value>
 	public virtual int NumberOfStates { get; internal set; } = 2;
 
+	/// <summary>
+	/// Modify this if you want your builder toggle have custom ordering.
+	/// You can specify which BuilderToggle to sort before/after
+	/// </summary>
+	public virtual Position OrderPosition { get; internal set; } = new Default();
+
 	/// <summary>
 	/// This is the current state of this builder toggle. Every time the toggle is clicked, it will change.<para/>
 	/// The default state is 0. The state will be saved and loaded for the player to be consistent.
 	/// </summary>
-	public int CurrentState => Main.player[Main.myPlayer].builderAccStatus[Type];
+	public int CurrentState
+	{ 
+		get => Main.LocalPlayer.builderAccStatus[Type]; 
+		set => Main.LocalPlayer.builderAccStatus[Type] = value; 
+	}
 
 	/// <summary>
 	/// This is the overlay color that is drawn on top of the texture.
```

Comment:
`DisplayColorTexture()` is good to obsolete with `[Obsolete(error: true)]`.
(To call such a hook, a non-erroring `[Obsolete]` wrapper method will be needed.)